### PR TITLE
Time picker fix

### PR DESCRIPTION
**Pull request by jun-w-huang**
Fixed bug where updating time in time picker would select the wrong time as it did not correctly account for timezone differences.

### CODE REVIEW

File: src/components/ControlledTimePicker.tsx [hunk 1]
```diff
@@ -67,20 +55,20 @@ const ControlledTimePicker = (props: ControlledTimePickerProps) => {
             minuteStep={15}
             use12Hours={true}
             value={displayedTime}
-            onBlur={fieldProps.onBlur}
             onSelect={(date) => {
-              setDisplayedTime(dayjs(date.valueOf()));
-              const convertedDate = convertInputDateToUTC(date.toDate());
-              fieldProps.onChange(convertedDate);
+              if (!date.isUTC()) {
```
Comment (devashish9) [inline src/components/ControlledTimePicker.tsx:59]: Remind me, where would a non-UTC date come from? Also if a UTC extension exist, does an EST extension exist as well?

File: src/components/ControlledTimePicker.tsx [hunk 2]
```diff
@@ -1,63 +1,51 @@
...
+const ControlledTimePickerRHF = (props: ControlledTimePickerRHFProps) => {
+  const [displayedTime, setDisplayedTime] = useState<Dayjs | undefined>(
+    props.value ? dayjs.utc(props.value) : undefined
+  );
 
   useEffect(() => {
     if (props.value) {
-      setDisplayedTime(dayjs.utc(props.value));
+      const utcOffset = props.value.getTimezoneOffset();
+      setDisplayedTime(dayjs(props.value).add(utcOffset, "minute"));
```
Comment (devashish9) [inline src/components/ControlledTimePicker.tsx:22]: Would it not be possible to do the same thing as line 16 here?

Comment (jun-w-huang) [reply src/components/ControlledTimePicker.tsx]: yes idk why i did it this way previously. just changed in new commit.

**Review by devashish9 — APPROVED** [review]
Looks good!

**Pull request merged**